src/Analyzers/Core/Analyzers/UseCollectionInitializer/AbstractObjectCreationExpressionAnalyzer.cs

```diff
@@ -24,166 +23,116 @@ internal abstract class AbstractObjectCreationExpressionAnalyzer<
+            State = state;
             _objectCreationExpression = objectCreationExpression;
             _analyzeForCollectionExpression = analyzeForCollectionExpression;
-            _cancellationToken = cancellationToken;
         }
 
         protected void Clear()
         {
-            _semanticModel = null;
-            _syntaxFacts = null;
+            State = default;
             _objectCreationExpression = null;
             _analyzeForCollectionExpression = false;
-            _cancellationToken = default;
-            _containingStatement = null;
-            _valuePattern = default;
-            _initializedSymbol = null;
         }
 
-        protected ImmutableArray<TMatch> AnalyzeWorker()
+        protected ImmutableArray<TMatch> AnalyzeWorker(CancellationToken cancellationToken)
         {
-            if (!ShouldAnalyze())
+            if (!ShouldAnalyze(cancellationToken))
                 return default;
 
-            _containingStatement = _objectCreationExpression.FirstAncestorOrSelf<TStatementSyntax>();
-            if (_containingStatement == null)
-                return default;
-
-            if (!TryInitializeVariableDeclarationCase() &&
-                !TryInitializeAssignmentCase())
-            {
-                return default;
-            }
-
             using var _ = ArrayBuilder<TMatch>.GetInstance(out var matches);
-            if (!TryAddMatches(matches))
+            if (!TryAddMatches(matches, cancellationToken))
                 return default;
 
             return matches.ToImmutable();
         }
 
-        private bool TryInitializeVariableDeclarationCase()
+        protected static UpdateExpressionState<TExpressionSyntax, TStatementSyntax>? TryInitializeState(
+            SemanticModel semanticModel,
+            ISyntaxFacts syntaxFacts,
+            TExpressionSyntax rootExpression,
+            CancellationToken cancellationToken)
         {
-            if (!_syntaxFacts.IsLocalDeclarationStatement(_containingStatement))
-            {
-                return false;
-            }
+            var statement = rootExpression.FirstAncestorOrSelf<TStatementSyntax>()!;
+            if (statement is null)
+                return null;
 
-            if (_objectCreationExpression.Parent.Parent is not TVariableDeclaratorSyntax containingDeclarator)
-            {
-                return false;
-            }
+            return
+                TryInitializeVariableDeclarationCase(semanticModel, syntaxFacts, rootExpression, statement, cancellationToken) ??
+                TryInitializeAssignmentCase(semanticModel, syntaxFacts, rootExpression, statement, cancellationToken);
+        }
 
-            _initializedSymbol = _semanticModel.GetDeclaredSymbol(containingDeclarator, _cancellationToken);
-            if (_initializedSymbol is ILocalSymbol local &&
+        private static UpdateExpressionState<TExpressionSyntax, TStatementSyntax>? TryInitializeVariableDeclarationCase(
```

Comment:
these helpers changed to be static.  instead of mutating in place, they now return the immtuable 'state' instance that is then used from that poitn on.